libs/execution_context/ebpf_link.c

```diff
@@ -97,6 +114,16 @@ ebpf_link_attach_program(ebpf_link_t* link, ebpf_program_t* program)
     link->program = program;
     ebpf_object_acquire_reference((ebpf_object_t*)program);
 
+    return_value = ebpf_program_get_properties(program, &program_parameters);
+    if (return_value != EBPF_SUCCESS) {
+        goto Done;
+    }
+
+    if (memcmp(&program_parameters.program_type, &link->program_type, sizeof(link->program_type)) != 0) {
```

Comment:
Minor: Since these are GUIDs, may be just a `==` is sufficient?

Reply:
Nope:
https://docs.microsoft.com/en-us/cpp/error-messages/compiler-errors-1/compiler-error-c2088

C doesn't support == for structs.